pkg/registry/core/shoot/strategy_test.go

```diff
@@ -25,6 +28,73 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 )
 
+var _ = Describe("Strategy", func() {
+	Context("PrepareForUpdate", func() {
+		Context("pod pids limit", func() {
+			It("should enforce the minimum limit value", func() {
+				var tooSmallValue int64 = 10
+
+				shoot := newShoot("foo")
+				shoot.Spec.Kubernetes.Kubelet = &core.KubeletConfig{
+					PodPIDsLimit: &tooSmallValue,
+				}
+				shoot.Spec.Provider.Workers = []core.Worker{
+					{
+						Kubernetes: &core.WorkerKubernetes{
+							Kubelet: &core.KubeletConfig{
+								PodPIDsLimit: &tooSmallValue,
+							},
+						},
+					},
+				}
+
+				oldShoot := newShoot("foo")
+				oldShoot.Spec.Kubernetes.Kubelet = shoot.Spec.Kubernetes.Kubelet.DeepCopy()
+				oldShoot.Spec.Provider.Workers = []core.Worker{shoot.Spec.Provider.Workers[0]}
+
+				shootregistry.Strategy.PrepareForUpdate(context.TODO(), shoot, oldShoot)
+
+				Expect(*shoot.Spec.Kubernetes.Kubelet.PodPIDsLimit).To(Equal(validation.PodPIDsLimitMinimum))
+				Expect(*oldShoot.Spec.Kubernetes.Kubelet.PodPIDsLimit).To(Equal(validation.PodPIDsLimitMinimum))
+				Expect(*shoot.Spec.Provider.Workers[0].Kubernetes.Kubelet.PodPIDsLimit).To(Equal(validation.PodPIDsLimitMinimum))
+				Expect(*oldShoot.Spec.Provider.Workers[0].Kubernetes.Kubelet.PodPIDsLimit).To(Equal(validation.PodPIDsLimitMinimum))
+			})
+
+			It("should not touch values that are above the minimum", func() {
+				var (
+					tooSmallValue   int64 = 10
+					highEnoughValue int64 = 105
+				)
+
+				shoot := newShoot("foo")
+				shoot.Spec.Kubernetes.Kubelet = &core.KubeletConfig{
+					PodPIDsLimit: &tooSmallValue,
+				}
+				shoot.Spec.Provider.Workers = []core.Worker{
+					{
+						Kubernetes: &core.WorkerKubernetes{
+							Kubelet: &core.KubeletConfig{
+								PodPIDsLimit: &highEnoughValue,
+							},
+						},
+					},
+				}
+
+				oldShoot := newShoot("foo")
+				oldShoot.Spec.Kubernetes.Kubelet = shoot.Spec.Kubernetes.Kubelet.DeepCopy()
+				oldShoot.Spec.Provider.Workers = []core.Worker{shoot.Spec.Provider.Workers[0]}
+
+				shootregistry.Strategy.PrepareForUpdate(context.TODO(), shoot, oldShoot)
+
+				Expect(*shoot.Spec.Kubernetes.Kubelet.PodPIDsLimit).To(Equal(validation.PodPIDsLimitMinimum))
+				Expect(*oldShoot.Spec.Kubernetes.Kubelet.PodPIDsLimit).To(Equal(validation.PodPIDsLimitMinimum))
+				Expect(*shoot.Spec.Provider.Workers[0].Kubernetes.Kubelet.PodPIDsLimit).To(Equal(*shoot.Spec.Provider.Workers[0].Kubernetes.Kubelet.PodPIDsLimit))
```

Comment:
```suggestion
				Expect(*shoot.Spec.Provider.Workers[0].Kubernetes.Kubelet.PodPIDsLimit).To(Equal(*highEnoughValue))
```
I guess this check is the same as `0 == 0`... 😅